src/stubs/ml_canvas.js

```diff
@@ -691,14 +717,20 @@ function ml_canvas_create_onscreen(autocmmit, decorated, resizeable, minimize,
   surface.id = "s" + id;
   surface.width = width;
   surface.height = height;
-  surface.style.position = "absolute"
-  surface.style.top = header_height + "px";
   surface.onmousedown = _surface_down_handler;
   surface.canvas = canvas;
   canvas.surface = surface;
   frame.appendChild(surface);
 
-  var ctxt = surface.getContext("2d");
+  if (resizeable === true) {
+    surface.style.width = "100%";
+    surface.style.height = "100%";
+    _resize.observe(surface);
+  }
+
+  // willReadFrequently needed to avoid warning on getImageData
+  var ctxt = surface.getContext("2d", {willReadFrequently: true});
```

Comment:
According to the doc, this force usage of a software renderer:
https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/getContext#willreadfrequently